src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java

```diff
@@ -168,6 +168,12 @@ public static synchronized ArrayList getFifoNextSpanData(
     } else {
       startTimeTrackerMs = FileAccessManager.readFileToString(timeMarkerFileName);
       startTimeTrackerMsLong = Long.parseLong(startTimeTrackerMs);
+      // If the file is corrupted, holds no data
+      if (startTimeTrackerMs.equalsIgnoreCase("empty")) {
```

Comment:
wouldnt line 169 be setting startTimeTrackerMs to null in the case where there isnt valid data?

Reply:
I changed this to a draft because it needed to be tested, you're right the exception is thrown when a null or empty value is being parse as a long. Whenever I upload an empty file to the device that file gets deleted, so it has been difficult to test.

Reply:
Sorry, didnt see the draft, just saw the review request.

Reply:
this is now ready for review.